src/algorithms/local_search/local_search.cpp

```diff
@@ -113,6 +113,173 @@ LocalSearch<Route,
     });
 }
 
+
+struct RouteInsertion {
+  Gain cost;
+  Index single_rank;
+  Index pickup_rank;
+  Index delivery_rank;
+};
+const RouteInsertion empty_insert = {std::numeric_limits<Gain>::max(), 0, 0, 0};
+
+template <class Route>
+RouteInsertion compute_best_insertion_single(const Input& input, const Index j, Index v, const Route& route) {
+  const auto& matrix = input.get_matrix();
+  RouteInsertion result = empty_insert;
+  const auto& current_job = input.jobs[j];
+  const auto& v_target = input.vehicles[v];
+
+  if (!input.vehicle_ok_with_job(v, j)) {
+      return result;
+  }
+
+  for (Index rank = 0; rank <= route.size(); ++rank) {
```

Comment:
In order to avoid an early `return` branch and only return at the end of the function, I'd be in favor of putting the `for` loop into an `if` with the opposite condition.